frameless-runtime/src/amoeba.rs

```diff
@@ -183,3 +183,307 @@ impl Verifier for AmoebaCreation {
+            generation: 2,
+            four_bytes: *b"test",
+        };
+        let input_data = vec![mother.into()];
+        let output_data = vec![d1.into(), d2.into()];
+
+        assert_eq!(
+            AmoebaMitosis.verify(&input_data, &output_data),
+            Ok(0),
+        );
+    }
+
+    #[test]
+    fn mitosis_wrong_generation() {
+        let mother = AmoebaDetails {
+            generation: 1,
+            four_bytes: *b"test",
+        };
+        let d1 = AmoebaDetails {
+            generation: 3, // This daughter has the wrong generation
+            four_bytes: *b"test",
+        };
+        let d2 = AmoebaDetails {
+            generation: 2,
+            four_bytes: *b"test",
+        };
+        let input_data = vec![mother.into()];
+        let output_data = vec![d1.into(), d2.into()];
+
+        assert_eq!(
+            AmoebaMitosis.verify(&input_data, &output_data),
+            Err(VerifierError::WrongGeneration),
+        );
+    }
+
+    #[test]
+    fn mitosis_badly_typed_input() {
+        let mother = Bogus;
+        let d1 = AmoebaDetails {
+            generation: 2,
+            four_bytes: *b"test",
+        };
+        let d2 = AmoebaDetails {
+            generation: 2,
+            four_bytes: *b"test",
+        };
+        let input_data = vec![mother.into()];
+        let output_data = vec![d1.into(), d2.into()];
+
+        assert_eq!(
+            AmoebaMitosis.verify(&input_data, &output_data),
+            Err(VerifierError::BadlyTypedInput),
+        );
+    }
+
+    #[test]
+    fn mitosis_no_input() {
+        let d1 = AmoebaDetails {
+            generation: 2,
+            four_bytes: *b"test",
+        };
+        let d2 = AmoebaDetails {
+            generation: 2,
+            four_bytes: *b"test",
+        };
+        let input_data = Vec::new();
+        let output_data = vec![d1.into(), d2.into()];
+
+        assert_eq!(
+            AmoebaMitosis.verify(&input_data, &output_data),
+            Err(VerifierError::WrongNumberInputs),
+        );
+    }
+
+    #[test]
+    fn mitosis_badly_typed_output() {
+        let mother = AmoebaDetails {
+            generation: 1,
+            four_bytes: *b"test",
+        };
+        let d1 = AmoebaDetails {
+            generation: 2,
+            four_bytes: *b"test",
+        };
+        let d2 = Bogus;
+        let input_data = vec![mother.into()];
+        let output_data = vec![d1.into(), d2.into()];
+
+        assert_eq!(
+            AmoebaMitosis.verify(&input_data, &output_data),
+            Err(VerifierError::BadlyTypedOutput),
+        );
+    }
+
+    #[test]
+    fn mitosis_only_one_output() {
+        let mother = AmoebaDetails {
+            generation: 1,
+            four_bytes: *b"test",
+        };
+        let d1 = AmoebaDetails {
+            generation: 2,
+            four_bytes: *b"test",
+        };
+        let input_data = vec![mother.into()];
+        // There is only one daughter when there should be two
```

Comment:
Similar to this comment I like this comment because it explains why you have this error type which can be seen in multiple types of scenarios.